lib/ts/querier.ts

```diff
@@ -180,8 +193,21 @@ export class Querier {
     // path should start with "/"
     sendGetRequest = async (
         path: NormalisedURLPath,
-        params: Record<string, boolean | number | string | undefined>
+        params: Record<string, boolean | number | string | undefined>,
+        userContext: any
     ): Promise<any> => {
+        const sortedKeys = Object.keys(params).sort();
+        let uniqueKey = path.getAsStringDangerous();
+
+        for (const key of sortedKeys) {
+            const value = params[key];
+            uniqueKey += `;${key}=${value}`;
+        }
+
+        if (uniqueKey in (userContext._default?.coreCallCache ?? {})) {
+            return userContext._default.coreCallCache[uniqueKey];
+        }
```

Comment:
Oh, I thought you'd keep the core call cache you had (where you set/get the cached object on the function impl level) and only added an invalidation logic here. Does it work the same way? 
I prefer the other method with invalidation added here as a safety precaution. Not sure if it's worth moving back to it though.

Reply:
we can discuss this